### PR TITLE
Video Streaming

### PR DESCRIPTION
**Pull request by TomaszTB**
Add video streaming to reduce network usage of sending image messages. 

As of now, only streaming color video results in less network usage. Future plans include converting 16 bit depth to RGB depth so depth can be streamed as color too. 

A comparison of compression algorithms is available [here](https://docs.google.com/spreadsheets/d/1v7t4t3xsZevzB9EKQWXs4v0Eg7EgHXgmC8PIESOF2oI/edit?usp=sharing)

Working on my desktop, laptop, and old Lenovo ThinkPad!

https://github.com/user-attachments/assets/7d2f66e6-ec9b-443b-832a-75ea7a33ba91

### CODE REVIEW

File: ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/logging/FFMPEGTools.java
Comment (TomaszTB) [inline ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/logging/FFMPEGTools.java:1]: Moved this guy to a new FFMPEG package in `ihmc-perception`. Perhaps a few other FFMPEG classes should follow?

File: ihmc-perception/src/main/java/us/ihmc/perception/ffmpeg/FFMPEGWebcamEncodingDemo.java
Comment (TomaszTB) [inline ihmc-perception/src/main/java/us/ihmc/perception/ffmpeg/FFMPEGWebcamEncodingDemo.java:1]: I made a lot of demos during development to make sure things work. I'm willing to delete a bunch of them if it's excessive. Just let me know how many to leave.

File: ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java [hunk 1]
```diff
@@ -285,4 +287,20 @@ public static ROS2Topic<BipedalSupportPlanarRegionParametersMessage> getBipedalS
...
+   // Realsense
+   public static final ROS2Topic<SRTStreamStatus> REALSENSE_STREAM_STATUS = STREAM_STATUS.withPrefix(D455_NAME);
+   public static final ROS2Topic<SRTStreamStatus> REALSENSE_COLOR_STREAM = REALSENSE_STREAM_STATUS.withSuffix("color");
+   public static final ROS2Topic<SRTStreamStatus> REALSENSE_DEPTH_STREAM = REALSENSE_STREAM_STATUS.withSuffix("depth");
+
+   // ZED
+   public static final ROS2Topic<SRTStreamStatus> ZED_STREAM_STATUS = STREAM_STATUS.withPrefix(ZED2_NAME);
+   public static final ROS2Topic<SRTStreamStatus> ZED_LEFT_COLOR_STREAM = ZED_STREAM_STATUS.withSuffix("color_left");
+   public static final ROS2Topic<SRTStreamStatus> ZED_RIGHT_COLOR_STREAM = ZED_STREAM_STATUS.withSuffix("color_right");
+   public static final ROS2Topic<SRTStreamStatus> ZED_DEPTH_STREAM = ZED_STREAM_STATUS.withSuffix("depth");
```
Comment (ds58) [inline ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java]: Can you prefix these with SRT_ ie `SRT_STREAM_STATUS`, `SRT_REALSENSE_STREAM_STATUS`, etc

**Review by ds58 — CHANGES_REQUESTED** [review]

File: ihmc-perception/src/main/resources/us/ihmc/perception/ffmpeg/example.webm
Comment (ds58) [inline ihmc-perception/src/main/resources/us/ihmc/perception/ffmpeg/example.webm:1]: Need to make sure this is in LFS

File: ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java (hunk 1 above)
Comment (TomaszTB) [reply ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java]: Done!

File: ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java [hunk 2]
```diff
@@ -285,4 +290,26 @@ public static ROS2Topic<BipedalSupportPlanarRegionParametersMessage> getBipedalS
...
    }
+
+   /* VIDEO STREAMING STUFF */
+   public static final ROS2Topic<?> STREAMING_MODULE = BEST_EFFORT.withModule(STREAMING_NAME);
+   public static final ROS2Topic<SRTStreamStatus> SRT_STREAM_STATUS = STREAMING_MODULE.withType(SRTStreamStatus.class);
+
+   // Realsense
+   public static final ROS2Topic<SRTStreamStatus> SRT_REALSENSE_STREAM_STATUS = SRT_STREAM_STATUS.withPrefix(D455_NAME);
+   public static final ROS2Topic<SRTStreamStatus> SRT_REALSENSE_COLOR_STREAM_STATUS = SRT_REALSENSE_STREAM_STATUS.withSuffix("color");
+   public static final ROS2Topic<SRTStreamStatus> SRT_REALSENSE_DEPTH_STREAM_STATUS = SRT_REALSENSE_STREAM_STATUS.withSuffix("depth");
```
Comment (calvertdw) [inline ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java:301]: Can you condense these into a reduced set of topics? I'd assume color and depth status could be communicated in the base status message/topic. This simplifies a lot of things, including reducing state inconsistency and threading errors.

File: ihmc-communication/src/main/java/us/ihmc/communication/packets/MessageTools.java [hunk 3]
```diff
@@ -1574,4 +1577,50 @@ else if (intLevel == Level.TRACE.intLevel())
...
          return Level.INFO;
    }
+
+   /**
+    * Serializes the given ROS2 message/
+    * @param message The ROS2 message to be serialized
+    * @return A ByteBuffer containing the serialized message
+    * @param <T> The type of the ROS2 message
+    */
+   public static <T extends Packet<T>> ByteBuffer serialize(T message)
```
Comment (calvertdw) [inline ihmc-communication/src/main/java/us/ihmc/communication/packets/MessageTools.java:1587]: Looks like these guys allocate and copy memory. Is that a performance penalty?

File: ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java (hunk 2 above)
Comment (TomaszTB) [reply ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java:301]: I'm not sure how I'd go about doing that. Do you have any immediate ideas?

Comment (calvertdw) [reply ihmc-communication/src/main/java/us/ihmc/communication/PerceptionAPI.java:301]: Nope we talked about this and I think it's good :+1:

**Review by ds58 — APPROVED** [review]

**Pull request merged**